### PR TITLE
add roll paddle frontend manipulation

### LINKED ISSUE #19236: roll

### PR DESCRIPTION
**Pull request by NripeshN**
close #19236

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by NripeshN** [comment]
add_frontend_checklist

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hi, thanks a ton for looking into this! :smile: 
Could you please update the supported [dtypes](https://github.com/PaddlePaddle/Paddle/blob/4f9404a6354f5442f4316c968356dd5aeec69c54/python/paddle/tensor/manipulation.py#L1715)?

Arg name should be `shifts` - https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/roll_en.html

Feel free to re-request a review after making those changes!

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hey! There was a recent addition of backend_to_test arg in the testing pipeline, which was causing the code to fail tests. I added that, but it's throwing `AttributeError: module 'paddle.fluid.dygraph' has no attribute 'ParamBase'` now. It doesn't look to be due to your code, so I've dropped a note on our [discord](https://discord.com/channels/799879767196958751/1028267758028337193/1133662065974837289) regarding this. :smile: 
Until it's fixed -
- Could you take care of the merge conflicts ?

**Comment by rishabgit** [comment]
Update: looks like that testing issue was resolved by [this](https://github.com/unifyai/ivy/commit/0ea0f555fc3a989d2d2329c3a62110e7112c9b4c)
Could you please update your branch with latest changes?

**Comment by NripeshN** [comment]
Hi @rishabgit, I have fixed all merge conflicts

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hey, your code looks great, but I'm getting an odd `AssertionError: returned shape = (), ground-truth returned shape = (1,)`. 
I've dropped a note to the testing team in case there's something off with ivy's testing pipeline. Could you please look into it in case it's not?

**Comment by NripeshN** [comment]
Hi @rishabgit, could you please review this PR

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hey, feel free to re-request review after changes. Thanks! smile

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py [hunk 1]
```diff
@@ -562,3 +562,34 @@ def test_paddle_flip(
...
+    dtype_and_x,
+    dtype,
+    on_device,
+    fn_tree,
+    frontend,
+    test_flags,
+    backend_fw,
+):
+    input_dtype, x = dtype_and_x
+    shifts = 1
```
[the remark this answers is not in the document]
Comment (NripeshN) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py]: Hi @rishabgit, I have made the necessary changes!

**Comment by rishabgit** [comment]
Thanks for the changes, they look great!

**Pull request merged**